lib/WorseReflection/Core/Inference/Walker/FunctionLikeWalker.php

```diff
@@ -71,7 +72,8 @@ private function walkFunctionLike(FrameResolver $resolver, Frame $frame, Functio
             ClassDeclaration::class,
             InterfaceDeclaration::class,
             TraitDeclaration::class,
-            EnumDeclaration::class
+            EnumDeclaration::class,
+            ObjectCreationExpression::class, // For Inline classes
```

Comment:
not sure this will work as object creation expressions can be inside class declarations etc.

Reply:
I have looked at the node tree. And where a normal definition has a ClassDefinitionNode there is an ObjectCreactionExpression

Reply:
ah i see, i'm confusing ancestor and descendant :facepalm: